stratum/hal/lib/tdi/tdi_sde_table_entry.cc

```diff
@@ -139,9 +141,12 @@ ::util::Status TdiSdeWrapper::DeleteTableEntry(
   device->createTarget(&dev_tgt);
 
   const auto flags = ::tdi::Flags(0);
-  RETURN_IF_TDI_ERROR(table->entryDel(
-      *real_session->tdi_session_, *dev_tgt, flags, *real_table_key->table_key_))
-      << "Could not delete table entry with: " << dump_args();
+  tdi_status_t status = table->entryDel(
```

Comment:
Are you sure that this is the correct behavior in all cases?

It is not necessarily wrong to treat a key-not-found condition as a success when you're deleting a key. The caller is asserting that there should be no table entry with the specified key, and that condition is true.

On the other hand, if the caller is expected to have verified the existence of the key before issuing the Delete, failure would reasonably be considered an internal error.

If the answer is "it depends", then perhaps this method should return ::util::error::Code::NOT_FOUND and allow the caller to make the decision.

(Apologies if I've already asked this.)